Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -257,24 +257,25 @@ public IEnumerable<EventDto> GetUserEvents(string username)
         return userEvents.ToDtoList().AsEnumerable();
     }
 
-    private string CreateSlug(EventDto eventDto, DateTime createdTime)
+    private string CreateSlug(EventDto eventDto)
     {
-        var slug = eventDto.Name.ToLower();
-        slug = slug.Trim();
-        slug = Regex.Replace(slug, @"\s+", " ");
-        slug = Regex.Replace(slug, " ", "-");
-        slug = Regex.Replace(slug, @"[^a-z0-9-]", "");
-        slug = Regex.Replace(slug, "-+", "-");
+        var baseSlug = eventDto.Name.ToLower();
+        baseSlug = baseSlug.Trim();
+        baseSlug = Regex.Replace(baseSlug, @"\s+", " ");
+        baseSlug = Regex.Replace(baseSlug, " ", "-");
+        baseSlug = Regex.Replace(baseSlug, @"[^a-z0-9-]", "");
+        baseSlug = Regex.Replace(baseSlug, "-+", "-");
 
-        var timestamp = createdTime.Ticks.ToString();
-        var maxSlugLength = 128 - timestamp.Length - 1; // -1 for the dash between slug and timestamp
+        var slug = baseSlug;
+        var counter = 1;
 
-        if (slug.Length > maxSlugLength)
+        while (context.Events.Any(e => e.Slug == slug))
         {
-            slug = slug.Substring(0, maxSlugLength);
+            slug = $"{baseSlug}-{counter}";
+            counter++;
         }
 
-        return $"{slug}-{timestamp}";
+        return slug;
     }

```

Comment:
Do as we agreed so get the counter from regex like getting `slug-{number}` or `slug` exactly and count from that, then `count + 1`, substring the `baseSlug` to be `128-(length of count)+1` and then do `baseSlug-count` and you have then 1 query instead of n, where n is number of events sharing those slugs


ref: https://stackoverflow.com/questions/66212794/how-to-match-regular-expression-on-a-column-for-postgresql-in-ef-core

Reply:
created